pkg/apis/serving/v1beta1/inference_service_status.go

```diff
@@ -383,3 +386,98 @@ func (ss *InferenceServiceStatus) ClearCondition(conditionType apis.ConditionTyp
 		conditionSet.Manage(ss).ClearCondition(conditionType)
 	}
 }
+
+func (ss *InferenceServiceStatus) UpdateModelRevisionStates(modelState ModelState, info *FailureInfo) {
+	if ss.ModelStatus.ModelRevisionStates == nil {
+		ss.ModelStatus.ModelRevisionStates = &ModelRevisionStates{ActiveModelState: modelState}
+	} else {
+		ss.ModelStatus.ModelRevisionStates.ActiveModelState = modelState
+	}
+	// update transition status based on model state
+	if modelState == Pending || modelState == Loading {
+		ss.ModelStatus.TransitionStatus = InProgress
+	} else if modelState == Loaded {
+		ss.ModelStatus.TransitionStatus = UpToDate
+	} else if modelState == FailedToLoad {
+		ss.ModelStatus.TransitionStatus = BlockedByFailedLoad
+	}
+	// update model failure info
+	if info != nil {
+		ss.SetModelFailureInfo(info)
+	}
+}
+
+func (ss *InferenceServiceStatus) UpdateModelTransitionStatus(status TransitionStatus, info *FailureInfo) {
+	ss.ModelStatus.TransitionStatus = status
+	// update model state
```

Comment:
nit: can we update this comment to something more helpful? The function name `UpdateModelTransitionStatus` is clear enough to know its a model update in my opinion. If we want a comment here, maybe something like
```
// if the model transition status is invalid, we need to update the model revision state and/because .... which will ...
```

Reply:
kind of similar to the comment on line 427. That comment is helpful and adds good context.

Reply:
@alexagriffith updated, Thanks!